services/QuillLMS/client/app/bundles/PremiumHub/containers/diagnosticGrowthReports/index.tsx

```diff
@@ -177,11 +197,59 @@ export const DiagnosticGrowthReportsContainer = ({
     setDownloadButtonBusy(true)
 
     requestPost('/admin_diagnostic_reports/download', {}, (body) => {
-      setShowSnackbar(true)
+      showInSnackbar("You will receive an email with a download link shortly.")
       setTimeout(() => { setDownloadButtonBusy(false) }, buttonDisableTime);
     })
   }
 
+  function handleClickSubscribe() {
+    setIsSubscriptionModalOpen(!isSubscriptionModalOpen);
+  }
+
+  function handleSubscriptionCancel() {
+    setIsSubscriptionModalOpen(false);
+  }
+
+  function handleSubscriptionSave(isSubscribed, frequency) {
+    setIsSubscriptionModalOpen(false);
+    if (isSubscribed) {
+      requestPost('/admin_report_filter_selections/show', { report: `${FILTER_SELECTIONS_REPORT_BASE.slice(0,-1)}` }, (sharedSelections) => {
+        requestPost('/admin_report_filter_selections/show', { report: `${FILTER_SELECTIONS_REPORT_BASE}${OVERVIEW}` }, (overviewSelections) => {
+          requestPost('/admin_report_filter_selections/show', { report: `${FILTER_SELECTIONS_REPORT_BASE}${SKILL}` }, (skillSelections) => {
+            requestPost('/admin_report_filter_selections/show', { report: `${FILTER_SELECTIONS_REPORT_BASE}${STUDENT}` }, (studentSelections) => {
+              if (sharedSelections) postFilterSelections(`subscription`, sharedSelections.filter_selections)
+              if (overviewSelections) postFilterSelections(`subscription_${OVERVIEW}`, overviewSelections.filter_selections)
+              if (skillSelections) postFilterSelections(`subscription_${SKILL}`, skillSelections.filter_selections)
+              if (studentSelections) postFilterSelections(`subscription_${STUDENT}`, studentSelections.filter_selections)
+
+              createOrUpdateEmailSubscription(frequency)
+            })
+          })
+        })
+      })
+
+    } else if (currentEmailSubscription) {
+      deleteEmailSubscription()
+    }
+    showInSnackbar('Subscription settings saved')
+  }
```

Comment:
I am not super-happy with this approach: fetching a bunch of API calls and then re-writing the data in them to another place if they exists.  However, I couldn't come up a with a cleaner way of doing this off-hand since on page-load the tabs that aren't active haven't actually loaded this data (so it's not in any state).  Would welcome suggestions on ways to clean this up or make it less... icky.